### PR TITLE
PPP Draft for Yit Hien

### PR DESCRIPTION
**Pull request by YitHien**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1,6 +1,24 @@
 # Developer Guide
 
+## Table of Contents
```
Comment (wli-linda) [inline docs/DeveloperGuide.md:3]: Thanks for adding this! I feel like there might be an automatic way of generating TOCs, but maybe not worth it to look into at this point.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -52,7 +70,14 @@ faster than a typical mouse driven or GUI driven apps.
 |v2.0|user|parse modules from NUSMods|easily add all my modules with a single command|
 |v2.0|user|save and load entries from a session|reload my entries from the previous session|
 
-## Design & implementation
+## Non-Functional Requirements
+1. Application should work on any mainstream operating system that has **Java 11** or above
```
Comment (wli-linda) [inline docs/DeveloperGuide.md:74]: Good addition!

File: src/main/java/seedu/allonus/expense/ExpenseTracker.java [hunk 3]
```diff
@@ -97,9 +97,12 @@
     public static final String KEYWORD_EDIT = "edit";
     public static final String KEYWORD_FIND = "find";
     public static final String KEYWORD_BLANK = "";
+    public static final String LOG_EMPTY_FIELD = "User input is missing a field";
+    public static final String LOG_EXTRA_FIELDS = "User enter multiple copies of the same delimiter";
+    public static final String LOG_INVALID_SLASH = "User entered slash in incorrect format";
 
     private static void expenseWelcome() {
-        System.out.println(EXPENSE_WELCOME_MESSAGE);
+        TextUi.showToUser(EXPENSE_WELCOME_MESSAGE);
```
Comment (wli-linda) [inline src/main/java/seedu/allonus/expense/ExpenseTracker.java:105]: Nice, thanks for making the formatting consistent!

**Review by wli-linda — COMMENTED** [review]
LGTM!

File: docs/diagrams/ExpenseTrackerSequenceDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+activate ExpenseTracker
+
+loop keep taking in user inputs for Expense module
+    ExpenseTracker -> ExpenseTracker : executeAdd(rawInput:String)
+    activate ExpenseTracker
+    deactivate ExpenseTracker
+
+    ExpenseTracker -> ExpenseParser : parseNewExpense(String)
+
+    activate ExpenseParser
```
Comment (OzairHasan) [inline docs/diagrams/ExpenseTrackerSequenceDiagram.puml:23]: Very good diagrams

**Review by OzairHasan — COMMENTED** [review]
LGTM

**Pull request merged**